### ISSUE TITLE
Have Travis CI push doc pages?

### ISSUE THREAD

**Issue by ceedubs** [non/cats#155]
Thanks to the excellent work of @stew and @tpolecat, we now have fancy documentation at http://non.github.io/cats/. It may be tough for us to remember to keep it up to date. Would it be possible/desired to have the Travis CI job for the master branch automatically publish those pages?

I'm not sure if this is tough from a permissions perspective. Also I don't know if there might be an issue with a really noisy git history for the pages that gets bulky when you pull the project?

**Comment by travisbrown** [comment]
I think it's perfectly reasonable to purge the gh-pages history occasionally if it gets too bulky, and it's not too hard to set up Travis to publish the site automatically.

The only concern I'd have about this is that when you have Travis doing something that relies on the central repo's `GH_TOKEN`, that means any builds on forks are going to fail. I'm sure there's some way to script your way out of this, but I've run into the issue on other projects, and if there's a nice solution I haven't seen it.

**Comment by non** [comment]
It sounds like @InTheNow is going to look into this.

@travisbrown do you feel strongly that we shouldn't set it up from Travis?

**Comment by travisbrown** [comment]
No, going ahead sounds good to me!—at a glance it looks like it shouldn't be too hard to set this up with `deploy` or `after_success` to get around my concern.

**Comment by rossabaker** [comment]
http4s publishes from Travis.  We call this [bash script](https://github.com/http4s/http4s/blob/v0.7.0/bin/travis-publish) from after_success.  I don't think that's a drop-in solution for cats, but it's a starting point.

**Comment by InTheNow** [comment]
@rossabaker thanks for the head start link and @travisbrown for the tip :)

**Comment by rossabaker** [comment]
The trickiest thing I encountered implementing this is to not prune docs from prior releases.  Help may be on the way with sbt/sbt-ghpages#10, but until then, I had to resort to [copying code out of the plugin](https://github.com/http4s/http4s/blob/059e0381628eb205b2c440c9146955582374c69a/project/Http4sGhPages.scala).

The API version `TaskKey[(Int, Int)]`, which reflects the current version minus the patch number.  I use that to specify the api subdirectory to git clean, and leave the remainder of api unscathed.  (This is also a useful key should we want to get into the business of MiMa checking for binary compatibility.)
